airbyte-integrations/connectors/source-braintree-singer/source_braintree_singer/source.py

```diff
@@ -0,0 +1,86 @@
+"""
+MIT License
+
+Copyright (c) 2020 Airbyte
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+import json
+import os
+import tempfile
+from datetime import datetime
+
+import braintree
+from airbyte_protocol import AirbyteConnectionStatus, Status
+from base_python import AirbyteLogger, ConfigContainer
+from base_singer import SingerSource
+from braintree.exceptions.authentication_error import AuthenticationError
+from dateutil import parser
+from dateutil.relativedelta import relativedelta
+
+TAP_CMD = "tap-braintree"
+
+DISCOVER_CONFIG_FILE = os.path.join(tempfile.gettempdir(), "discover_configs.json")
+
+
+class SourceBraintreeSinger(SingerSource):
+    def transform_config(self, raw_config):
+        # The tap-braintree singer moves the start_date 1 month earlier, this line fixes this.
+        raw_config["start_date"] = (parser.parse(raw_config["start_date"]) + relativedelta(months=+1)).strftime("%Y-%m-%dT%H:%M:%SZ")
```

Comment:
just to make sure I understand, braintree syncs data from the month previous to the input date? so if I input 2020-08-01 (august 1st) it starts syncing from july 1st?

Reply:
exactly